de.dashup.application/src/main/webapp/views/marketplace.jsp

```diff
@@ -7,17 +9,166 @@
+                </div>
+            </div>
+        </nav>
+        <form action="${fn:escapeXml(pageContext.request.contextPath)}/marketplace/search" method="post">
+            <div class="row" style="margin-top: 25px">
+                <div class="col m4 offset-m4 s8 offset-s1">
+                    <div class="input-field">
+                        <input id="text-field-search-term" name="email" type="text" class="validate">
+                        <label for="text-field-search-term"><fmt:message key="i18n.enterSearchterm"/></label>
+                    </div>
+                </div>
+                <div class="col m2 s2" style="margin-top: 18px">
+                    <button id="btn_start_search_marketplace" class="btn waves-effect waves-light" type="submit"
+                            name="search">
+                        <fmt:message key="i18n.go"/>
+                    </button>
+                </div>
+            </div>
+        </form>
+        <div class="row">
+            <div class="col m6 offset-m3 s10 offset-s1">
+                <h3><fmt:message key="i18n.featured"/></h3>
+                <hr/>
+            </div>
+        </div>
+        <div class="row">
+            <div class="col m6 offset-m3 s10 offset-s1">
+                <div class="carousel carousel-slider center">
+                    <c:forEach items="${featuredWidgets}" var="mapEntry">
+                        <a style="display:block" href="${pageContext.request.contextPath}/marketplace/detailView/${fn:escapeXml(mapEntry.key.id)}">
+                            <div class="carousel-item" href="#one!" style="background-color: var(--color-secondary);color: #ffffff">
+                                <div class="row" style="margin: 10px">
+                                    <div class="col">
+                                        <i class="fas fa-${fn:escapeXml(mapEntry.key.iconCode)} fa-7x"></i>
+                                    </div>
+                                    <div class="col">
+                                        <h1 style="margin-top: 20px">${fn:escapeXml(mapEntry.key.name)}</h1>
+                                    </div>
+                                </div>
+                                <div class="row" style="margin-left: 10px; margin-right: 10px;">
+                                    <p>
+                                        ${fn:escapeXml(mapEntry.key.shortDescription)}
+                                    </p>
+                                </div>
+                                <div class="row" style="margin: 0 auto; width: 164px">
+                                    <div class="col">
+                                        <div class="star-rating" style="font-size: 25px; margin-top: 15px; color: var(--color-dark-gray);">
+                                            <div class="back-stars">
+                                                <i class="fa fa-star" aria-hidden="true"></i>
```

Comment:
You could use the jsp for loop here

Reply:
I can use the jsp loop to iterate over a list or map or something but not like `for(int i= 0;i<5;i++)`.
Feel free to inform yourself here: https://www.tutorialspoint.com/jsp/jsp_standard_tag_library.htm.
In addition to that I would not do a for loop here cause its just 2 lines shorter and not that readable.

Reply:
Using JSLT you can only use forEach on pageContext objects. But you can as well use scriptlets, printing out markup to the JSP view or adding new pageContext objects. Did it for you.

Reply:
http://www.jsptut.com/Scriptlets.jsp
Here you go